### PR TITLE
Pickles: random number of fixed tables and lookup in these tables

### PR DESCRIPTION
**Pull request by dannywillems**
Explain your changes:
* Built on top of https://github.com/MinaProtocol/mina/pull/14052.
* Same than 14052 but with more lookups and more tables

Explain how you tested your changes:
* `dune exec src/lib/pickles/test/optional_custom_gates/pickles_test_optional_custom_gates.exe`

Checklist:

- [ ] Dependency versions are unchanged
  - Notify Velocity team if dependencies must change in CI
- [ ] Modified the current draft of release notes with details on what is completed or incomplete within this project
- [ ] Document code purpose, how to use it
  - Mention expected invariants, implicit constraints
- [ ] Tests were added for the new behavior
  - Document test purpose, significance of failures
  - Test names should reflect their purpose
- [ ] All tests pass (CI will check this if you didn't)
- [ ] Serialized types are in stable-versioned modules
- [ ] Does this close issues? List them

* Closes #0000

### CODE REVIEW

**Comment by dannywillems** [comment]
!ci-build-me

File: src/lib/pickles/test/optional_custom_gates/pickles_test_optional_custom_gates.ml [hunk 1]
```diff
@@ -239,6 +239,53 @@ let main_fixed_lookup_tables () =
...
        } )
 
+(* Parameters *)
+(* nb of fixed lookup tables *)
+let m = Random.State.int state 10
+
+let f_lt_data =
+  Array.init m ~f:(fun _ ->
+      let size = Random.State.int state 100 in
+      let indexes = Array.init size ~f:(fun i -> Field.Constant.of_int i) in
```
Comment (rbonichon) [inline src/lib/pickles/test/optional_custom_gates/pickles_test_optional_custom_gates.ml:249]:
```suggestion
      let indexes = Array.init size ~f:(Field.Constant.of_int) in
```

File: src/lib/pickles/test/optional_custom_gates/pickles_test_optional_custom_gates.ml [hunk 2]
```diff
@@ -239,6 +239,53 @@ let main_fixed_lookup_tables () =
        ; (* v3 *) w6 = exists Field.typ ~compute:(fun () -> v3)
        } )
 
+(* Parameters *)
+(* nb of fixed lookup tables *)
```
Comment (rbonichon) [inline src/lib/pickles/test/optional_custom_gates/pickles_test_optional_custom_gates.ml:243]:
```suggestion
(* number of fixed lookup tables *)
```
Btw, why between 0 and 10? More specifically why 10 as an upper limit and why include 0 as a possibility?

File: src/lib/pickles/test/optional_custom_gates/pickles_test_optional_custom_gates.ml [hunk 3]
```diff
@@ -239,6 +239,53 @@ let main_fixed_lookup_tables () =
...
+      let size = Random.State.int state 100 in
+      let indexes = Array.init size ~f:(fun i -> Field.Constant.of_int i) in
+      let values =
+        Array.init size ~f:(fun _ ->
+            Field.Constant.of_int (Random.State.int state 100_000_000) )
+      in
+      (indexes, values) )
+
+(* nb of lookups *)
+let n = Random.State.int state 10
```
Comment (rbonichon) [inline src/lib/pickles/test/optional_custom_gates/pickles_test_optional_custom_gates.ml:257]: I assume the 0-10 questions might be the same here?

File: src/lib/pickles/test/optional_custom_gates/pickles_test_optional_custom_gates.ml [hunk 4]
```diff
@@ -239,6 +239,53 @@ let main_fixed_lookup_tables () =
...
+  Array.init m ~f:(fun _ ->
+      let size = Random.State.int state 100 in
+      let indexes = Array.init size ~f:(fun i -> Field.Constant.of_int i) in
+      let values =
+        Array.init size ~f:(fun _ ->
+            Field.Constant.of_int (Random.State.int state 100_000_000) )
+      in
+      (indexes, values) )
+
+(* nb of lookups *)
```
Comment (rbonichon) [inline src/lib/pickles/test/optional_custom_gates/pickles_test_optional_custom_gates.ml:256]:
```suggestion
(* number of lookups *)
```

**Review by rbonichon — APPROVED** [review]
Only minor questions, though I guess they're worth addressing. 

LGTM

File: src/lib/pickles/test/optional_custom_gates/pickles_test_optional_custom_gates.ml (hunk 2 above)
Comment (dannywillems) [reply src/lib/pickles/test/optional_custom_gates/pickles_test_optional_custom_gates.ml:243]: No strong reason.
The `0` values are fixed in another PR.

**Comment by dannywillems** [comment]
!ci-build-me

File: src/lib/pickles/test/optional_custom_gates/pickles_test_optional_custom_gates.ml (hunk 2 above)
Comment (dannywillems) [reply src/lib/pickles/test/optional_custom_gates/pickles_test_optional_custom_gates.ml:238]: Including the fix in this one already.

File: src/lib/pickles/test/optional_custom_gates/pickles_test_optional_custom_gates.ml (hunk 3 above)
Comment (dannywillems) [reply src/lib/pickles/test/optional_custom_gates/pickles_test_optional_custom_gates.ml]: See https://github.com/MinaProtocol/mina/pull/14056#discussion_r1319427471

**Comment by dannywillems** [comment]
!ci-build-me

**Pull request merged**